metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/utils/SyslogUtils.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.parsers.utils;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class SyslogUtils {
+
+    public static long convertToEpochMillis(String logTimestamp, String logTimeFormat) {
+        ZonedDateTime timestamp = ZonedDateTime.parse(logTimestamp, DateTimeFormatter.ofPattern(logTimeFormat).withZone(ZoneOffset.UTC));
+        return timestamp.toEpochSecond() * 1000;
+    }
+
+    public static long parseTimestampToEpochMillis(String logTimestamp) {
+        if (logTimestamp.length() < 20) {
+            ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
+            int year = now.getYear();
+            if (now.getDayOfYear() == 1 && !now.getMonth().toString().substring(0,3).equals(logTimestamp.substring(0,3).toUpperCase()))
+                year--;
```

Comment:
I like the idea of checking how far the date in the current year would be in the future and basing the back date decision on that. Let me work on that.